packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -115,18 +127,32 @@ class FlutterPlatform extends PlatformPlugin {
         processToKill: process, // This kills the subprocess whether coverage is enabled or not.
       );
 
+      Completer<Null> timeout = new Completer<Null>();
+
       // Pipe stdout and stderr from the subprocess to our printStatus console.
-      _pipeStandardStreamsToConsole(process);
+      _pipeStandardStreamsToConsole(process, handleLine: (String line) {
```

Comment:
put the string comparison logic in _pipeStandardStreamsToConsole alongside the "error: Unable to read Dart source" code. Just pass a callback for "main started" that does the future stuff.